micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -754,7 +754,7 @@ default <T> T scoped(Supplier<T> supplier) {
      * @return the result from {@link CheckedCallable#call()}
      */
     @SuppressWarnings("unused")
-    default <T, E extends Throwable> T scopedChecked(CheckedCallable<T, E> checkedCallable) throws E {
+    default <T, E extends Throwable> @Nullable T scopedChecked(CheckedCallable<T, E> checkedCallable) throws E {
```

Comment:
Hmm I'm not sure how this should be declared. I guess the nullability on the generic type `T` on `CheckedCallable` should match the return type nullability.

Reply:
I think this is the way. :)

Reply:
Should we do the same for `observe`, `observeChecked`, etc?